pkg/util/util.go

```diff
@@ -239,3 +244,184 @@ func SplitQualifiedName(str string) (string, string) {
 func JoinQualifiedName(namespace, name string) string {
 	return path.Join(namespace, name)
 }
+
+type Route struct {
+	Interface   string
+	Destination net.IP
+	Gateway     net.IP
+	// TODO: add more fields here if needed
+}
+
+func getRoutes(input io.Reader) ([]Route, error) {
+	routes := []Route{}
+	if input == nil {
+		return nil, fmt.Errorf("input is nil")
+	}
+	scanner := bufio.NewReader(input)
+	for {
+		line, err := scanner.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		if strings.HasPrefix(line, "Iface") {
+			continue
+		}
+		fields := strings.Fields(line)
+		routes = append(routes, Route{})
```

Comment:
I think that was a me-ism, and it was just because extra copies annoy me.
I don't much care either way.

On Tue, Mar 10, 2015 at 6:00 AM, Justin Santa Barbara <
notifications@github.com> wrote:

> In pkg/util/util.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4865#discussion_r26118728>
> :
>
> > +func getRoutes(input io.Reader) ([]Route, error) {
> > +	routes := []Route{}
> > +	if input == nil {
> > +		return nil, fmt.Errorf("input is nil")
> > +	}
> > +	scanner := bufio.NewReader(input)
> > +	for {
> > +		line, err := scanner.ReadString('\n')
> > +		if err == io.EOF {
> > +			break
> > +		}
> > +		if strings.HasPrefix(line, "Iface") {
> > +			continue
> > +		}
> > +		fields := strings.Fields(line)
> > +		routes = append(routes, Route{})
>
> I'm not familiar with this idiom. Is there a reason you don't just build
> the Route and then append it? (Efficiency?)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4865/files#r26118728>
> .
>